File: pkg/weighted-pool-factory/math-implementations/python/math_check.py
"""Maths implementation for weighted-pool-factory. See pvt/math-check/README.md for the interface."""
from weighted_pool import Weighted

# The `poolType` written by pkg/weighted-pool-factory/math-check/pool.ts.
POOL_TYPE = "WEIGHTED"


def create_pool(pool):
    return Weighted(pool)

File: pkg/weighted-pool-factory/math-implementations/rust/src/bin/math_check.rs
//! Runs this implementation against bush-maths test data: `cargo run --features math-check --bin math_check -- <file>`.

use bush_maths::pools::weighted::WeightedState;
use bush_maths::PoolBase;
use math_check_support::{no_hook, PoolJson};
use weighted_pool_maths::WeightedPool;

fn make_pool(pool: &PoolJson) -> Result<Box<dyn PoolBase>, String> {
    let state = WeightedState {
        base: pool.base_state(None)?,
        weights: pool.arr("weights")?,
        min_token_balances: pool.arr("minTokenBalances").ok(),
    };
    Ok(Box::new(WeightedPool::from(state)))
}

fn main() {
    math_check_support::run(make_pool, no_hook);
}
